### ISSUE TITLE
`PTH201` does not report redundant joins

### ISSUE THREAD

**Issue by tjkuson** [astral-sh/ruff#10453]
Using `path-constructor-current-directory` (`PTH201`),

```python
import pathlib


p = pathlib.Path(".") / "foo.txt"
```

is auto-fixed to

```python
import pathlib


p = pathlib.Path() / "foo.txt"
```

by invoking `ruff check --select PTH201 --fix --isolated /tmp/cwd.py` using version 0.3.2.

The fix is correct. However, it would be more useful if it included the rest of the path (following the motivation of the rule, which is to reduce unnecessary explicit CWD specification). For example,

```python
import pathlib


p = pathlib.Path("foo.txt")
```

Search terms: PTH201, current working directory, pathlib
